Make necessary PottsCell fields public, prepare for abstract setModule() function

Jannetty opened the pull request:
- added getters and setters for private PottsCell class fields that need to be accessed to implement default make function
- changed default make function to exclusively use getters and setters to access private fields
- added tests for new getters and setters
- broke setState into two functions: `setState `which assigns the Cell's state field and calls `setStateModule`, and `setStateModule` which assigns the cell's state module. Did not add or change any tests associated with this change (existing tests check if state field is correctly set and check if state module is correctly set)

These changes are done in anticipation of making PottsCell and abstract class. All subclasses will not have access to private fields so will need to access them through getters and setters. The default PottsCell implementation will use the exact make and setStateModule functions as they are defined in this version of the code. Future PottsCell implementations will define different make and setStateModule functions to accommodate different cellular behaviors.

jessicasyu commented:
I think it would be fine to just drop the `private` modifier for the fields you need? The same pattern is used in `patch`.

Jannetty commented:
- Changed fields accessed via getters and setters to be public.
- Deleted associated tests
- Left breaking of setState into two functions (so setStateModule can be abstract when PottsCell is abstract class)
